drivers/pcie/host/pcie.c

```diff
@@ -83,6 +83,7 @@ bool pcie_get_mbar(pcie_bdf_t bdf, unsigned int index, struct pcie_mbar *mbar)
 
```

Comment:
Ok I see why pcie_get_mbar() is not doing what is required for MSI-X BIR (which what triggered this patch to be created in the first place).

index != bar index. And that comes back from the first commit adding this function (commit id e0390535467)
Index is not a BAR index but a valide BA index.
And it's documented that way "The indices are order-preserving with respect to the endpoint BARs: e.g., index 0 will return the lowest-numbered memory BAR on the endpoint."

Let's take an example of 3 memory Base Address. Here in 32 bits only:

- BA 0 (BAR index 0)
- BA 1 (BAR index 1)
- BA 2 (BAR index 2)

Let's say now that BA 1 is in 64 bits, it will be:

- BA 0 (BAR index 0)
- BA 1 (BAR index 1)
- BA 2 (BAR index 3 <- because BAR index 2 is also part of BA 1 it's higher address part)

So that's what index represents here. BA x, not BAR index x.
And it's not practical when it comes to MSI-x with it's BIR: "Base address register Indicator Register". Indicator can be interpreted as Index as well.

So pcie_get_mbar() should be kept as is. The proposed patch is not valid (it break what pcie_get_mbar() is supposed to do, according to its documentation). 

I'll came a fix for msi-x BIR.